File: app/src/main/java/com/goranch/publicapis/ui/home/HomeModule.java
package com.goranch.publicapis.ui.home;

import com.goranch.publicapis.di.scopes.HomeScope;

import dagger.Module;
import dagger.Provides;

/**
 * Created by goran on 01/10/2016.
 */

@Module
public class HomeModule {

    private HomeView mHomeView;

    public HomeModule(HomeView homeView) {
        mHomeView = homeView;
    }

    @Provides
    @HomeScope
    HomeView provideHomeView() {
        return mHomeView;
    }

    @Provides
    @HomeScope
    HomePresenter provideHomePresenter(HomePresenter homeView) {
        return new HomePresenterImpl(mHomeView);
    }
}

File: app/src/main/java/com/goranch/publicapis/ui/home/HomeModule.kt
package com.goranch.publicapis.ui.home

import com.goranch.publicapis.di.scopes.HomeScope

import dagger.Module
import dagger.Provides

@Module
class HomeModule(private val mHomeView: HomeView) {

    @Provides
    @HomeScope
    internal fun provideHomeView(): HomeView {
        return mHomeView
    }

    @Provides
    @HomeScope
    internal fun provideHomePresenter(homeView: HomePresenter): HomePresenter {
        return HomePresenterImpl(mHomeView)
    }
}
